Updated menu

ykenichi opened the pull request:
Hey Allen, I added some code to the menu to select buttons from the menu using the arrow keys and the spacebar. I experimented with the buttons instead of using sprites, but I kind of like the keyboard method more. If you want buttons I can change it to use buttons instead of switching the sprites. Please let me know what you think.
- Kenichi

ykenichi committed:
Updated menu with buttons

ykenichi committed:
Added asset files

ackei commented:
Hi Kenichi,

Nice work so far. I'm amazed by the progress.

Before I merge, just a few things. For one, remove the npm log from your commits. You can probably git rm it from your local repo and push that. Then, you can add *.log to the .gitignore file to ignore any files ending with .log when adding and committing.

The button is up to you primarily. However, I do remember that the Phaser buttons have built in parameters for changing how the button looks when clicked, when mouse is hovering over, etc. That may be simpler to use when we want to include some nice button effects in the future. However, I leave that decision up to you =).

After the clean up, I'll check one more time and merge the code.

Thanks.

ykenichi committed:
Removed npm-debug.log

ykenichi committed:
Added placeholder sounds and title music.

ykenichi committed:
Added placeholder sounds and title music.

ykenichi commented:
Hey Allen, I removed the npm log and added .log to the .gitignore file. I think it should be fine now.

The reason why I made the menu work with keypresses and not mouse clicks is because I thought that since the game's controls would be primarily keyboard-based, it would be an inconvenience to the play to switch between the mouse and keyboard, especially if using a laptop. However, if you guys prefer I have no issues with changing it so that it works with a mouse click.

I was experimenting with the buttons and tried to get both the mouse-over functionality as well as the keyboard functionality working, however, since I had the menu selection code for the keyboard implemented in the update() function it would quickly override the mouse-over and click appearance (the menu hover and press operation would only be visible for 1 frame). I added a wrap-around to the menu to make it easier to navigate.

I also added some placeholder title music and menu button selection/switching code to the game. I got it from the phaser examples folder, and I made sure to use the update() function you made in Preloader.js so that it decodes the music before playing. If you guys don't like it please feel free to remove it.

If there are any changes you want me to make let me know and I'll push out another update.

- Kenichi

ackei commented:
Okay, I'll leave it up to you. Ideally, we will probably want the buttons to be clickable as well, but I'll merge it for now! Thanks for your help
